File: 08. Analisis Kompleksitas/B. Cek Bilangan Prima/b_cek_bilangan_prima.py
N = int(input())
for i in range(N):
    num = int(input())
    isPrime = True
    if (num == 1):
        isPrime = False
    else:
        j = 2
        while (j*j <= num):
            if (num % j == 0):
                isPrime = False
                break
            j += 1
    if (isPrime):
        print("YA")
    else:
        print("BUKAN")

File: 08. Analisis Kompleksitas/B. Cek Bilangan Prima/b_cek_bilangan_prima.rs
fn main() {
    let mut reader = String::new();
    std::io::stdin().read_line(&mut reader).expect("");
    let n: i32 = reader.trim().parse().expect("");
    for _ in 0..n {
        let mut reader = String::new();
        std::io::stdin().read_line(&mut reader).expect("");
        let num: i32 = reader.trim().parse().expect("");
        let mut is_prime: bool = true;
        if num == 1 {
            is_prime = false;
        } else {
            let mut j: i32;
            j = 2;
            while j * j <= num {
                if num % j == 0 {
                    is_prime = false;
                    break;
                }
                j += 1;
            }
        }
        if is_prime {
            println!("YA");
        } else {
            println!("BUKAN");
        }
    }
}
